### ISSUE TITLE
Nomad action from ui with job name does not work

### ISSUE THREAD

**Issue by sbenoistmics** [hashicorp/nomad#23464]
### Nomad version
Nomad v1.8.0
BuildDate 2024-05-28T17:38:17Z
Revision 28b82e4b2259fae5a62e2ed47395334bea5a24c4

### Operating system and Environment details
74-Ubuntu SMP Wed Feb 22 14:14:39 UTC 2023

### Issue
When you add a name stanza in your job, actions does not work when launched from ui
I guess that this should be job.get('id') and instead of this [job.get('name')](https://github.com/hashicorp/nomad/blob/cd3101d624f24a6c35bf7c2363b34b79e71ac7eb/ui/app/adapters/job.js#L195)
You can try to command the second line of the job in exemple to see the different behaviours

### Reproduction steps
Post job with name stanza, with an action, then run action from ui

#### Expected Result
You should see an output
It should tell `OCI runtime exec failed: exec failed: unable to start container process: exec: "ls": executable file not found in $PATH: unknown` with exemple job, but it is not what matters

#### Actual Result
The return of action say `Action completed with no output`

### Job file (if appropriate)

```
job "nomad-action-from-ui-with-job-name-does-not-work" {
  name        = "Nomad action from ui with job name does not work"
  region      = "FRA"

  group "gp" {
    task "tk" {
      driver = "docker"
      config {
        image = "hashicorp/http-echo"
        args  = ["-text", "hello world"]
      }

      resources {
        cpu = 20
      }

      action "ls" {
        command = "ls"
        args    = ["-a"]
      }
    }
  }
}
```

**Comment by philrenaud** [comment]
Thanks for the issue and great find re:
> I guess that this should be job.get('id') and instead of this [job.get('name')](https://github.com/hashicorp/nomad/blob/cd3101d624f24a6c35bf7c2363b34b79e71ac7eb/ui/app/adapters/job.js#L195)

Let me look into patching this as soon as possible. Glad to see you using Nomad Actions!

**Comment by philrenaud** [comment]
Thanks for the issue, @sbenoistmics — should be fixed in #23553 and released soon.
